listeners/eventhub.py

```diff
@@ -0,0 +1,22 @@
+
+class EventHub:
+    listener_dict = dict()
+
+    @classmethod
+    def subscribe(cls, key, listener):
+        if key in cls.listener_dict.keys():
+            cls.listener_dict[key].append(listener)
+        else:
+            cls.listener_dict[key] = [listener]
+
+    @classmethod
+    def run_listeners(cls, key, *args):
+        if key in cls.listener_dict.keys():
+            for handler in cls.listener_dict[key]:
+                handler(*args)
+
+    @classmethod
+    def run_listener_with_return(cls, key, *args):
```

Comment:
This doesn't seem right/safe to me.  What happens if multiple listeners sign-up for an event, and it is run via this method?  Only the first one will execute right?